docs/porting-example/steps/step_02_hpy_legacy.c

```diff
@@ -0,0 +1,173 @@
+#include <math.h>
+#include <Python.h>
+#include <hpy.h>
+
+// Porting to HPy, Step 2: Porting some methods
+//
+// An example of porting a C extension that implements a Point type
+// with a couple of simple methods (a norm and a dot product). It
+// illustrates the steps needed to port types that contain additional
+// C attributes (in this case, x and y).
+//
+// This file contains an example second step of the port in which some methods
+// have been converted to HPy methods that receive handles as arguments, but
+// other methods are still legacy methods that receive PyObject arguments.
+
+typedef struct {
+    // PyObject_HEAD is required while legacy methods still access
+    // PointObject and should be removed once the port to HPy is completed.
+    PyObject_HEAD
+    double x;
+    double y;
+    // HPy handles are shortlived to support all GC strategies
+    // For that reason, PyObject* in C structs are replaced by HPyField
+    HPyField obj;
+} PointObject;
+
+// This defines PyPointObject as an alias of PointObject so that existing
+// code that still uses PyPointObject and expects PyObject_HEAD continues to
+// compile and run. Once PyObject_HEAD has been removed, this alias should be
+// removed so that code that still expects PyObject_HEAD will fail to compile.
+typedef PointObject PyPointObject;
+
+// The legacy type helper macro defines an PointObject_AsStruct function allows
+// non-legacy methods to convert HPy handles to PointObject structs. The legacy
+// type helper macro is used because PyObject_HEAD is still present in
+// PointObject. Once PyObject_HEAD has been removed (see point_hpy_final.c) we
+// will use HPy_TYPE_HELPERS instead.
+HPyType_LEGACY_HELPERS(PointObject)
+
+HPyDef_SLOT(Point_traverse, Point_traverse_impl, HPy_tp_traverse)
+int Point_traverse_impl(void *self, HPyFunc_visitproc visit, void *arg)
+{
+    HPy_VISIT(&((PointObject*)self)->obj);
+    return 0;
+}
+
+// this is a method for creating a Point
+HPyDef_SLOT(Point_init, Point_init_impl, HPy_tp_init)
```

Comment:
Since we merged PR #367, the `HPyDef_*` macros changed. Again, not your fault. I'll update it (just adding this comment for reminding myself).